src/sources/aws_s3/sqs.rs

```diff
@@ -0,0 +1,733 @@
+    300
+}
+const fn default_true() -> bool {
+    true
+}
+
+#[derive(Debug, Snafu)]
+pub enum IngestorNewError {
+    #[snafu(display("Unable to fetch queue URL for {}: {}", name, source))]
+    FetchQueueUrl {
+        source: RusotoError<GetQueueUrlError>,
+        name: String,
+        owner: Option<String>,
+    },
+    #[snafu(display("Got an empty queue URL for {}", name))]
+    MissingQueueUrl { name: String, owner: Option<String> },
+    #[snafu(display("Invalid visibility timeout {}: {}", timeout, source))]
+    InvalidVisibilityTimeout {
+        source: std::num::TryFromIntError,
+        timeout: u64,
+    },
+}
+
+#[derive(Debug, Snafu)]
+pub enum ProcessingError {
+    #[snafu(display(
+        "Could not parse SQS message with id {} as S3 notification: {}",
+        message_id,
+        source
+    ))]
+    InvalidSqsMessage {
+        source: serde_json::Error,
+        message_id: String,
+    },
+    #[snafu(display("Failed to fetch s3://{}/{}: {}", bucket, key, source))]
+    GetObject {
+        source: RusotoError<GetObjectError>,
+        bucket: String,
+        key: String,
+    },
+    #[snafu(display("Failed to read all of s3://{}/{}: {}", bucket, key, error))]
+    ReadObject {
+        error: String,
+        bucket: String,
+        key: String,
+    },
+    #[snafu(display(
+        "Object notification for s3://{}/{} is a bucket in another region: {}",
+        bucket,
+        key,
+        region
+    ))]
+    WrongRegion {
+        region: String,
+        bucket: String,
+        key: String,
+    },
+}
+
+pub struct Ingestor {
+    region: Region,
+
+    s3_client: S3Client,
+    sqs_client: SqsClient,
+
+    multiline: Option<line_agg::Config>,
+    compression: super::Compression,
+
+    queue_url: String,
+    poll_interval: Duration,
+    visibility_timeout_secs: i64,
+    delete_message: bool,
+}
+
+impl Ingestor {
+    pub async fn new(
+        region: Region,
+        sqs_client: SqsClient,
+        s3_client: S3Client,
+        config: Config,
+        compression: super::Compression,
+        multiline: Option<line_agg::Config>,
+    ) -> Result<Ingestor, IngestorNewError> {
+        let queue_url_result = sqs_client
+            .get_queue_url(GetQueueUrlRequest {
+                queue_name: config.queue_name.clone(),
+                queue_owner_aws_account_id: config.queue_owner.clone(),
+                ..Default::default()
+            })
+            .await
+            .with_context(|| FetchQueueUrl {
+                name: config.queue_name.clone(),
+                owner: config.queue_owner.clone(),
+            })?;
+
+        let queue_url = queue_url_result
+            .queue_url
+            .ok_or(IngestorNewError::MissingQueueUrl {
+                name: config.queue_name.clone(),
+                owner: config.queue_owner.clone(),
+            })?;
+
+        // This is a bit odd as AWS wants an i64 for this value, but also doesn't want negative
```

Comment:
Ideas here welcome :smile:

Reply:
Alternatively, limit the timeout secs config to a `u32` and cast it to `i64` safely.

Reply:
:+1: good idea. It'll artificially limit the value, but I can't imagine anyone wanting to set a value higher than u32 anyway.